seslib/__init__.py

```diff
@@ -491,7 +491,7 @@ def _generate_static_networks(self):
             self.settings.cluster_network = cluster_network
 
     def _generate_nodes(self):
-        node_id = 1
+        node_id = 0
         for node_roles in self.settings.roles:
```

Comment:
```suggestion
        for node_id, node_roles in enumerate(self.settings.roles, start=1):
```
This is probably the better way to iterate with an index here. That should solve the list issues and the duplicate ip address.

Reply:
I think this will not work because we don't want to increment the `node_id` when role is `admin` or `suma`.